### PR TITLE
Make resource details sections more obvious

### LINKED ISSUE #1612: Add background to the expander headers in the new resource details panel
A follow up suggestions to #1542 : 
https://github.com/dotnet/aspire/pull/1542#issuecomment-1882893982
https://github.com/dotnet/aspire/pull/1542#issuecomment-1883240381

Add a background to the expander headers for better section visibility.

### PR DESCRIPTION
**Pull request by tlmii**
(Hopefully) resolves #1612 

I worked on this for a while and, given the relatively small color palette we have to work with, coming up with an accordion item header color that worked well didn't seem to be an easy task. So I went back to the drawing board a bit and this is what I came up with. I think this accomplishes the actual goal of making the individual sections much more easily identifiable. Though I acknowledge it doesn't do it in the specific way that the issue suggested, so I'm certainly open to feedback. 

Some screenshots of how it looks:

ResourceDetails_Dark_PanelOnly
ResourceDetails_Light_PanelOnly
ResourceDetails_Dark_FullPage
ResourceDetails_Light_FullPage
ResourceDetails_Dark_Collapsed
ResourceDetails_Light_Collapsed

This has the effect of somewhat resembling the way messages are displayed in new Outlook/outlook.com (though I left off the drop-shadow for now at least):
image

 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/dotnet/aspire/pull/2270)

### CODE REVIEW

**Comment by davidfowl** [comment]
Did we investigate tabs? They would be immediately visible on opening the details pane.

**Comment by tlmii** [comment]
I did look at tabs, in part because that's a somewhat common thing used in office in some of their right-side flyouts. From my perspective, that made it too many clicks to get to what you want, though, especially when details are on the right. As it stands now, environment variables are usually (mostly) visible on the screen or at worst with a flick of the mouse wheel. With tabs, you'd have to move the mouse to the environment variables tab and then click to see the same information.

I think if this list of details ends up getting a lot longer, we can look at tabs again but I don't think its a better experience at this point.

**Comment by kvenkatrajan** [comment]
We may want to consider short cut keys for expand/collapse of specific sections for keyboard accessbility navigation

CC: @adamint

**Review by kvenkatrajan — APPROVED** [review]

**Review by JamesNK — CHANGES_REQUESTED** [review]
image

This white is too white. It should be be off white.

In the Outlook example, the color is #FAFAFA.

image

Also, the text is too dark.

tldr; there is too much contrast

**Comment by JamesNK** [comment]
What about using the same color as the filter textbox:

image

The dark mode is doing that and it looks ok.

**Comment by tlmii** [comment]
I didn't pick the white, it's just the `--neutral-layer-1` value (next lighter from the `--neutral-layer-2` we use as the background in the main body) based on the color tokens the designers gave us. Same as in dark mode. 

I'd rather not fight the color system if possible. Let me see what I can do.

**Comment by JamesNK** [comment]
nit: The badge color around numbers has almost disappeared in dark mode and the same will happen in light mode if the background is make less white.

Change to a darker color, such as the darker background or black (although black would probably look bad):

image

**Comment by tlmii** [comment]
@JamesNK 
> This white is too white. It should be be off white.
> In the Outlook example, the color is #FAFAFA.

Technically the active message in the outlook example had a background of `#FFFFFF`, same as in my light original theme screenshots. The non-active ones are `#FAFAFA` as you mention.

Here's a screenshot of what it looks like using the same color as the textboxes for light theme only (and with the headers being less bold):

image

Personally I think the contrast between the background and the accordion items isn't quite enough in this case, but its similar to the inactive messages in Outlook, so I went with it for now.

Also adjusted the badge color on dark theme:

image

**Comment by JamesNK** [comment]
I like it.

Some things I noticed:
image

1. The header color covers the rounded corners. Setting the color needs to move from the header to the accordion item.
2. The button in the header (with up or down icon) is now lighter than the accordion background. It should be the same color or transparent in default state. The current hover color is fine.

image

Could you add a small margin to the bottom of the details view. Right now, if the bottom item reaches the bottom of the screen, then there is no margin.

**Comment by tlmii** [comment]
@JamesNK OK I think I got all that taken care of. There's some real oddities in the way the underlying CSS is laid out for the accordion (the (body) region has a color but the header doesn't, there's nested parts and you can't style something inside of a part, etc). Rounded corners should be back and non-hovered icon color should be correct.

Swapped out the existing left/right margins for padding left/right/bottom (there was already a gap at top) that I think addresses the bottom padding request.

**Review by JamesNK — APPROVED** [review]

**Review by drewnoakes — APPROVED** [review]
Love the indentation. Makes it much easier to segment quickly.
